src/ex2/pokemonClient.js

```diff
@@ -0,0 +1,43 @@
+
+class PokemonClient{
+    constructor(){}
+    async getListPokemonInfo(pokemonsIdList){
+        let pokemonsToReturn = [];
+        const allFetching = [];
+        try{
+            pokemonsIdList.forEach((pokemonId) => {
+                allFetching.push(fetch(`https://pokeapi.co/api/v2/pokemon/${pokemonId}`));
+        });
+            await Promise.all(allFetching).then(results => {
+                return Promise.all(results.map(response => response.json()));
+            }).then(pokemons => {
+                pokemons.forEach(pokemon => {
+                    let stringOfPokemonType = this.getPokemonTypes(pokemon);
+                    pokemonsToReturn.push(`Catch ${pokemon.name} the ${stringOfPokemonType} type pokemon`);
+                })
+            })
+            return pokemonsToReturn;
+        }catch (error) {
+            throw `Faild to fetch pokemon with this input : ${pokemonsIdList}`;
```

Comment:
Try to avoid scenarios where you concatenate a parameter to the error, but keep it clean. You're going to handle it the same way anyhow